modules/js-and-css/speculation-rules/settings.php

```diff
@@ -0,0 +1,218 @@
+	$eagerness_labels = plsr_get_eagerness_labels();
+
+	// Ensure only valid keys are present.
+	$value = array_intersect_key( $input, $default_value );
+
+	// Set any missing or invalid values to their defaults.
+	if ( ! isset( $value['mode'] ) || ! isset( $mode_labels[ $value['mode'] ] ) ) {
+		$value['mode'] = $default_value['mode'];
+	}
+	if ( ! isset( $value['eagerness'] ) || ! isset( $eagerness_labels[ $value['eagerness'] ] ) ) {
+		$value['eagerness'] = $default_value['eagerness'];
+	}
+
+	return $value;
+}
+
+/**
+ * Registers setting to control Speculation Rules configuration.
+ *
+ * @since n.e.x.t
+ * @access private
+ */
+function plsr_register_setting() {
+	register_setting(
+		'reading',
+		'plsr_speculation_rules',
+		array(
+			'type'              => 'object',
+			'description'       => __( 'Configuration for the Speculation Rules API.', 'performance-lab' ),
+			'sanitize_callback' => 'plsr_sanitize_setting',
+			'default'           => plsr_get_setting_default(),
+			'show_in_rest'      => array(
+				'schema' => array(
+					'properties' => array(
+						'mode'      => array(
+							'description' => __( 'Whether to prefetch or prerender URLs.', 'performance-lab' ),
+							'type'        => 'string',
+							'enum'        => array_keys( plsr_get_mode_labels() ),
+						),
+						'eagerness' => array(
+							'description' => __( 'Whether to trigger on click (conservative), on hover (moderate), or on even the slight suggestion (eager) that the user may navigate to the URL.', 'performance-lab' ),
+							'type'        => 'string',
+							'enum'        => array_keys( plsr_get_eagerness_labels() ),
+						),
+					),
+				),
+			),
+		)
+	);
+}
+add_action( 'init', 'plsr_register_setting' );
+
+/**
+ * Adds the settings sections and fields for the Speculation Rules configuration.
+ *
+ * @since n.e.x.t
+ * @access private
+ */
+function plsr_add_setting_ui() {
+	add_settings_section(
+		'plsr_speculation_rules',
+		__( 'Speculation Rules', 'performance-lab' ),
+		static function () {
+			?>
+			<p class="description">
+				<?php esc_html_e( 'This section allows you to control how URLs that your users navigate to are speculatively loaded to improve performance.', 'performance-lab' ); ?>
+			</p>
+			<?php
+		},
+		'reading'
+	);
+
+	$fields = array(
+		'mode'      => array(
+			'title'       => __( 'Mode', 'performance-lab' ),
+			'description' => __( 'Prerendering will lead to faster load times than prefetching, however in case of interactive content prefetching may be a safer choice.', 'performance-lab' ),
+		),
+		'eagerness' => array(
+			'title'       => __( 'Eagerness', 'performance-lab' ),
+			'description' => __( 'The eagerness setting defines the heuristics based on which the loading is triggered.', 'performance-lab' )
+				. '<br>' . __( '"Eager" is best from a performance perspective, "Conservative" increases the chance that only URLs the user actually navigates to are loaded.', 'performance-lab' ),
+		),
+	);
```

Comment:
@domenic @jeremyroman Would be great to get a quick check from you on the explanations in the UI (mostly here, also in a few other places in this file).

Reply:
"best" seems a loaded term. Yes it gives most chance of the prerendering being completed before use, but also risks causing other performance problems (e.g. wasted resources/bandwidth) that may have other impacts.

WDYT about something like this:

> "Eager" will have the minimum delay to start prerendering. "Conservative"...

Reply:
Thanks, that sounds good. Except I don't like to use "prerendering", since it's actually "prefetching or prerendering". I'm broadly using the term "loading" to cover both, since repeating these two terms everywhere is very verbose and long. I realize "loading" may not be the most adequate term, please let me know if there's a better single word to describe these two alternatives.

Reply:
Updated in https://github.com/WordPress/performance/pull/939/commits/b1399613364dcaa8df75f12582b37d840ac3da3b

Reply:
"Speculative loads" or "Speculations" is what DevTools uses (it used to use "Preloading" in the past but [we got complaints](https://x.com/tunetheweb/status/1718899373168341106?s=20)).

Reply:
Updated in https://github.com/WordPress/performance/pull/939/commits/24c3bb09c40e75bf8aa8927c6f9cd98dcda13a05.

I renamed "Mode" to "Speculation Mode" in the UI to clarify that relationship better.